applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -307,6 +309,15 @@ def InitializeSolutionStep(self):
         # TODO the response value calculation for stresses currently only works on the adjoint modelpart
         # this needs to be improved, also the response value should be calculated on the PRIMAL modelpart!!
         self.adjoint_analysis.time = self.adjoint_analysis._GetSolver().AdvanceInTime(self.adjoint_analysis.time)
+
+        # Put primal solution on adjoint model - for "auto" setting, else it has to be done by the user e.g. using hdf5 process
+        if self.response_settings["adjoint_settings"].GetString() == "auto":
+            Logger.PrintInfo("> Transfer primal state to adjoint model part.")
+            for primal_node, adjoint_node in zip(self.primal_model_part.Nodes, self.adjoint_model_part.Nodes):
+                for variable in self.primal_state_variables:
+                    values_to_transfer = primal_node.GetSolutionStepValue(variable)
+                    adjoint_node.SetSolutionStepValue(variable, values_to_transfer)
```

Comment:
Please use the function from variable-utils for this

Reply:
don't know why i always forget this. thanks!